kotlinx-coroutines-core/jvm/src/flow/TickerFlow.kt

```diff
@@ -0,0 +1,33 @@
+package kotlinx.coroutines.flow
+
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.awaitClose
+import java.util.*
+import kotlin.concurrent.schedule
+
+/**
+ * Creates a flow that produces the first item after the given initial delay and subsequent items with the
+ * given delay between them.
+ *
+ * The resulting flow is a callback flow, which basically listens @see [Timer.schedule]
+ *
+ * This Flow stops producing elements immediately after [Job.cancel] invocation.
+ *
+ * @param period period between each element in milliseconds.
+ * @param initialDelay delay after which the first element will be produced (it is equal to [period] by default) in milliseconds.
+ */
+public fun tickerFlow(
+        period: Long,
+        initialDelay: Long = period
+): Flow<Unit> = callbackFlow {
+    require(period > 0)
+    require(initialDelay > -1)
+
+    val timer = Timer()
+    timer.schedule(initialDelay, period) {
+        offer(Unit)
+    }
```

Comment:
This implementation uses a background thread and is JVM-specific.
Couldn't we simply use a loop with `delay()` and make it multiplatform? Or am I naive here?

Reply:
Thanks! very good point.
 Also `ticker` implementation can be moved to `commonMain`.

Reply:
I can see now why a concurrent behaviour is necessary and would like to backtrack a bit: we can't use a simple loop with delays because otherwise the ticker delay is impacted by the execution time of the collector code, which is probably not desirable in this case.

That's why we do need some concurrent process (a coroutine?) to send the ticks.
This still doesn't mean we have to use a thread and Java's timer.
There may be a multiplatform way to do this.

I'll let the maintainers of the lib comment on this, though